### DISCUSSION TITLE
Minitest and FactoryDefault

### DISCUSSION THREAD

**Discussion by davidwessman** [test-prof/test-prof#299] [Q&A]
Hello!

I would really like to investigate how to use more of the tools in test-prof, but we are not using RSpec and it seems like a lot of examples are for RSpec today.

I am currently investigating if we could use `FactoryDefault` with Minitest and I have tried to added something similar to the RSpec recipe:

```ruby
if defined?(TestProf::BeforeAll)
  TestProf::BeforeAll.configure do |config|
    config.before(:begin) do |context|
      TestProf::FactoryDefault.current_context = context
    end

    config.after(:rollback) do |context|
      TestProf::FactoryDefault.reset(context: context)
    end
  end
end
```

But it seems like there is only the `before_all` method available after including it like [this](https://github.com/test-prof/test-prof/blob/master/docs/recipes/before_all.md#minitest).

Have anyone figured out a way to do it?

**Comment by palkan** [comment]
I think, another bit that is required is to set and reset the context per example (test method in Minitest): https://github.com/test-prof/test-prof/blob/b891e72de01efde9e099c538ed79f665dfc9655f/lib/test_prof/recipes/rspec/factory_default.rb#L20-L25

So, you need to do something similar via `#before_setup` and `#after_teardown` methods.
